handler_test.go

```diff
@@ -603,6 +605,45 @@ func (h *testHandlers) M3Post(arg *struct {
 	h.c.Check(arg.P, gc.Equals, 99)
 }
 
+func (*handlerSuite) TestHandlersRootFuncWithRequestArg(c *gc.C) {
+	handleVal := testHandlers{
+		c: c,
+	}
+	var gotArg interface{}
+	f := func(p httprequest.Params, arg interface{}) (*testHandlers, context.Context, error) {
```

Comment:
I'd like to see a test with a compatible arg that isn't the empty interface please.